src/core/device.c

```diff
@@ -441,10 +441,7 @@ static int device_add_udev_wants(Unit *u, sd_device *dev) {
                 }
         }
 
-        strv_free(d->wants_property);
-        d->wants_property = TAKE_PTR(added);
-
-        return 0;
+        return strv_free_and_replace(d->wants_property, added);
```

Comment:
maybe split this out into its own PR so tht we can merge this before this PR is settled? this looks uncontroversial after all

Reply:
Moved to #17057.